packages/eslint-plugin/src/rules/dependency-checks.ts

```diff
@@ -120,32 +127,19 @@ export default createESLintRule<Options, MessageIds>({
       return {};
     }
 
-    // gather helper dependencies for @nx/js executors
-    const helperDependencies = getHelperDependenciesFromProjectGraph(
-      workspaceRoot,
-      sourceProject.name,
-      projectGraph
-    );
-
     const rootPackageJson = getPackageJson(join(workspaceRoot, 'package.json'));
 
-    // find all dependencies for the project
-    const npmDeps = findProjectsNpmDependencies(
+    const npmDependencies = findNpmDependencies(
+      workspaceRoot,
       sourceProject,
       projectGraph,
-      buildTarget,
-      rootPackageJson,
+      projectFileMap,
+      buildTarget, // TODO: What if child library has a build target different from the parent?
```

Comment:
We would have to list all "build" targets. But it's very unlikely that nomenclature diverges between projects

Reply:
We assume that there is just a single build target. This needs to be fixed, but we can do this is a followup PR